src/App.tsx

```diff
@@ -1,26 +1,223 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { deleteTodo, postTodo, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
+import { FilterStatus } from './types/FilterStatus';
+import { TodoList } from './components/TodoList/TodoList';
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>(todosFromServer);
+  const [shownTodos, setShownTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [todosFilter, setTodosFilter] = useState(FilterStatus.All);
+  const [selectedTodo, setSelectedTodo] = useState<Todo>();
+  const [todoToDelete, setTodoToDelete] = useState<Todo | null>(null);
+  const [shouldDeleteCompleted, setShouldDeleteCompleted] = useState(false);
+  const [newTodo, setNewTodo] = useState<Todo | null>(null);
+  const [todoBeingAdded, setTodoBeingAdded] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [clearInput, setClearInput] = useState(false);
+  const focusedTodoRef = useRef<HTMLInputElement>(null);
+  const defaultInputRef = useRef<HTMLInputElement>(null);
+
+  function showError(errMessage: string) {
```

Comment:
You must follow one way of creating functions
```suggestion
 const showError = (errMessage: string) => {
```